docs/oauth/oauth-single-sign-on.mdx

```diff
@@ -0,0 +1,306 @@
+
+This example is written for Next.js App Router but can be adapted for any framework/language. It assumes the OAuth flow has already been completed and that you have obtained an access token from Clerk.
+
+```tsx {{ filename: 'app/api/userinfo/route.ts' }}
+import type { NextRequest } from 'next/server'
+import { NextResponse } from 'next/server'
+
+export async function GET() {
+  // Assume you already have the access token available here
+  const accessToken = 'YOUR_ACCESS_TOKEN_HERE'
+
+  try {
+    const response = await fetch(`${process.env.FAPI_URL}/oauth/userinfo`, {
+      headers: {
+        // Include the access token in the Authorization header
+        Authorization: `Bearer ${accessToken}`,
+      },
+    })
+
+    if (!response.ok) {
+      const errorData = await response.json()
+      return NextResponse.json(errorData, { status: response.status })
+    }
+
+    const data = await response.json()
+    return NextResponse.json(data)
+  } catch (error: unknown) {
+    return NextResponse.json(
+      { error: 'Failed to fetch user info', details: (error as Error).message },
+      { status: 500 },
+    )
+  }
+}
+```
+
+**Example response**
+
+```json
+{
+  "object": "string",
+  "instance_id": "string",
+  "user_id": "string",
+  "sub": "string",
+  "email": "string",
+  "email_verified": true,
+  "family_name": "string",
+  "given_name": "string",
+  "name": "string",
+  "username": "string",
+  "preferred_username": "string",
+  "picture": "string",
+  "public_metadata": {},
+  "private_metadata": {},
+  "unsafe_metadata": {}
+}
+```
+
+The `/oauth/userinfo` endpoint provides the following user properties, depending on the granted scopes:
+
+| Property | Description |
+| - | - |
+| `user_id` | The ID of the user |
+| `sub` | The ID of the user |
+| `given_name` | The user's first name |
+| `family_name` | The user's last name |
+| `name` | The user's full name |
+| `picture` | The user's avatar URL |
+| `preferred_username` | The user's username |
+| `email` | The user's primary email address |
+| `email_verified` | Whether the user's primary email address is verified |
+| `public_metadata` | The user's public metadata |
+| `private_metadata` | The user's private metadata |
+| `unsafe_metadata` | The user's unsafe metadata |
+
+### Get token information
+
+For validating access tokens or refresh tokens and retrieving additional token metadata, Clerk provides a standard OAuth 2.0 [Token Introspection Endpoint](https://www.oauth.com/oauth2-servers/token-introspection-endpoint/) at [`/oauth/token_info`](/docs/reference/frontend-api/tag/OAuth2-Identify-Provider#operation/getOAuthTokenInfo){{ target: '_blank' }}.
```

Comment:
Second link didn't exist, first link was to an okta-owned site, prefer the spec itself

```suggestion
For validating access tokens or refresh tokens and retrieving additional token metadata, Clerk provides a standard OAuth 2.0 [Token Introspection Endpoint](https://datatracker.ietf.org/doc/html/rfc7662) at [`/oauth/token_info`](https://clerk.com/docs/reference/frontend-api/tag/OAuth2-Identify-Provider#operation/getOAuthTokenInfo){{ target: '_blank' }}.
```